README.md

```diff
@@ -228,6 +229,55 @@ jobs:
 
 > **_NOTE:_** Building Storybook locally makes it simple to test Storybooks that could be available remotely, but are under authentication layers. If you also deploy your Storybooks somewhere (e.g. Chromatic, Vercel, etc.), the Storybook URL can still be useful with the test-runner. You can pass it to the `REFERENCE_URL` environment variable when running the test-storybook command, and if a story fails, the test-runner will provide a helpful message with the link to the story in your published Storybook instead.
 
+## Experimental test hook API
+
+The test runner renders a story and executes its [play function](https://storybook.js.org/docs/react/writing-stories/play-function) if one exists. However, there are certain behaviors that are not possible to achieve via the play function, which executes in the browser. For example, if you want the test runner to take visual snapshots for you, this is something that is possible via Playwright/Jest, but must be executed in Node.
+
+To enable use cases like visual or DOM snapshots, the test runner exports test hooks that can be overridden globally. These hooks give you access to the test lifecycle before and after the story is rendered.
+
+Consider the following pseudocode:
+
+```js
+it('component--widget', async () => {
+  const page = newPage();
+  const context = { id: 'component--widget', title: 'Component', name: 'Widget' };
+  await page.goto(STORYBOOK_URL);
+
+  // pre-render hook
+  if (preRender) await preRender(page, context);
+
+  // render the story and run its play function (if applicable)
+  await page.execute('render', context);
+
+  // post-render hook
+  if (postRender) await postRender(page, context);
+});
+```
+
+The hooks here, `preRender` and `postRender` are functions that take a [Playwright Page](https://playwright.dev/docs/pages) and a context object with the current story `id`, `title`, and `name`. They are globally settable by `@storybook/test-runner`'s `setPreRender` and `setPostRender` APIs.
+
+Thus, to make the test runner perform image snapshotting, you might set up the following in your `jest-setup.js`:
```

Comment:
I think we need to add a better description here.

People need to:
1 - "Eject" the config file
2 - Add a [setupFileAfterEnv](https://jestjs.io/docs/configuration#setupfilesafterenv-array) section
3 - Add the jest-setup.js (or whatever other name) file to the project.

Would it make sense to have some sort of "eject" command? So we generate the file override for the users instead?

I'm fine with not doing any of that and just adding a link to the jest setup file docs, but the average user might get lost in the instructions

Reply:
Eject command sounds great!

Reply:
Can you handle it @yannbf ? https://github.com/storybookjs/test-runner/issues/42

Reply:
Sure!